tensorflow/python/ops/array_ops.py

```diff
@@ -1917,7 +1917,6 @@ def one_hot(indices, depth, on_value=1, off_value=0,
             axis, dtype], name, "one_hot") as name:
     on_value = ops.convert_to_tensor(on_value, dtype=dtype, name="on_value")
```

Comment:
@ebrevdo When these are switched to `cast`, we end up with an `UnimplementedError` when trying to cast standard Python numbers to `complex64`, while `convert_to_tensor` has code coverage for this. Might we stick with `convert_to_tensor` for now and switch after the appropriate `cast` implementations are covered? From what I understand, the main downside of `convert_to_tensor` is that it may not work properly when `on_value` and `off_value` are passed in as Tensor objects, but I imagine most people just passing in normal Python types.